internal/database/statement.go

```diff
@@ -110,3 +110,17 @@ func SqlairClauseAnd(columnValues map[string]any) (string, sqlair.M) {
 	}
 	return strings.Join(terms, " AND "), args
 }
+
+// Prepare prepares a SQLair query. If the query has been prepared previously it
+// is retrieved from the statement cache.
+type StatementBase interface {
```

Comment:
What is the reasoning for putting this here rather than in `domain/state.go`?

Reply:
Interfaces should be defined at use (ideally).

Reply:
What about the internal/data/base/interface.go file?

Reply:
I think he's talking about all of the additions including implementation. Comment is wrong too.